common/version.go

```diff
@@ -0,0 +1,106 @@
+package common
+
+import (
+	"runtime"
+	"os"
+	"strings"
+	"os/user"
+	"github.com/denisbrodbeck/machineid"
+	"strconv"
+	"math/rand"
+	"time"
+	"fmt"
+)
+
+var clientSessionIdentity *ClientSessionIdentityT
+var SessionID string // global because we use in mlog fns to inject for all data points
+
+func init() {
+	SetClientSessionIdentity("")
+}
+
+// ClientSessionIdentityT holds values describing the client, environment, and session.
+type ClientSessionIdentityT struct {
+	Version string `json:"version"`
+	Hostname string `json:"host"`
+	Username string `json:"user"`
+	MachineID string `json:"machineid"`
+	Goos string `json:"goos"`
+	Goarch string `json:"goarch"`
+	Goversion string `json:"goversion"`
+	Pid int `json:"pid"`
+	SessionID string `json:"session"`
+	StartTime time.Time `json:"start"`
+}
+
+// String is the stringer fn for ClientSessionIdentityT
+func (s *ClientSessionIdentityT) String() string {
+	return fmt.Sprintln(s.Goos, s.Goarch, s.SessionID, s.Hostname, s.Username, s.MachineID, strconv.Itoa(s.Pid))
+}
+
+// Helpers for random sessionid string.
+const letterBytes = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+func randStringBytes(n int) string {
+	b := make([]byte, n)
+	for i := range b {
+		b[i] = letterBytes[rand.Intn(len(letterBytes))]
+	}
+	return string(b)
+}
+
+// SetClientSessionIdentity sets the global variable describing details about the client and session.
+// It is idempotent, excepting only being a setter for Version value.
+func SetClientSessionIdentity(versionName string) {
+	if clientSessionIdentity != nil {
+		clientSessionIdentity.Version = runtime.Version()
+		return
+	}
+
+	rand.Seed(time.Now().UnixNano())
+	SessionID = randStringBytes(4)
```

Comment:
Just a small note: I changed this to length `4` instead of original `8` for legibility and simplicity.. thinking 52^4 should be plenty of randomness for nearly every use case.